unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -628,4 +634,93 @@ public int hashCode() {
     }
     return result;
   }
+
+  /**
+   * Encodes a string into a Soundex value. Soundex is an encoding used to relate similar names,
+   * but can also be used as a general purpose scheme to find word with similar phonemes.
+   * https://en.wikipedia.org/wiki/Soundex
+   */
+  public UTF8String soundex() {
+    if (numBytes == 0) {
+      return UTF8String.fromBytes(new byte[0]);
```

Comment:
EMPTY_UTF8 = UTF8String.fromString(""); Use this or the old one will not impact the functionality and performance. Chenghao, function, logic bugs and major performance degradation are the 3 most important things in the software development. Even if we care about the Nic like blank string, lines at this time, the code will be rewritten in the future. We just need reformat code before release automatically by a code in build.